api/maven-api-core/src/main/java/org/apache/maven/api/Constants.java

```diff
@@ -462,6 +462,15 @@ public final class Constants {
     @Config(type = "java.lang.Boolean", defaultValue = "true")
     public static final String MAVEN_CONSUMER_POM = "maven.consumer.pom";
 
+    /**
+     * User property for controlling "maven modern personality". If set to {@code false} Maven will behave
+     * as previous major version.
+     *
+     * @since 4.0.0
+     */
+    @Config(type = "java.lang.Boolean", defaultValue = "true")
+    public static final String MAVEN_MODERN_PERSONALITY = "maven.modernPersonality";
```

Comment:
Would it be more intuitive to have a config 'maven3compatibility' (I don't like this naming either) with a default value to 'false' ?

Reply:
Agreed, but went with `maven.maven3Personality` as Maven4 is per def "compatible" with Maven3, this is more like "run Maven4 fully in Maven3 mode".